src/einsteinpy/symbolic/weyl.py

```diff
@@ -0,0 +1,96 @@
+    """
+
+    def __init__(self, arr, syms, config="llll", parent_metric=None):
+        """
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+            Sympy Array or multi-dimensional list containing Sympy Expressions
+        syms : tuple or list
+            Tuple of crucial symbols dentoting time-axis, 1st, 2nd, and 3rd axis (t,x1,x2,x3)
+        config : str
+            Configuration of contravariant and covariant indices in tensor. 'u' for upper and 'l' for lower indices. Defaults to 'llll'.
+        parent_metric : ~einsteinpy.symbolic.metric.WeylTensor
+            Corresponding Metric for the Weyl Tensor. Defaults to None.
+
+        Raises
+        ------
+        TypeError
+            Raised when arr is not a list or sympy Array
+        TypeError
+            syms is not a list or tuple
+        ValueError
+            config has more or less than 3 indices
+
+        """
+        super(WeylTensor, self).__init__(arr, config=config)
+        self.syms = syms
+        self._order = 4
+        self._parent_metric = parent_metric
+        if isinstance(syms, (list, tuple)):
+            self.syms = syms
+            self.dims = len(self.syms)
+        else:
+            raise TypeError("syms should be a list or tuple")
+        if not len(config) == self._order:
+            raise ValueError("config should be of length {}".format(self._order))
+
+    @classmethod
+    def from_metric(cls, metric, dims, parent_metric=None):
+        """
+        Get Weyl tensor calculated from a metric tensor
+
+        Parameters
+        ----------
+        metric : ~einsteinpy.symbolic.metric.MetricTensor
+            Space-time Metric from which Christoffel Symbols are to be calculated
+        dims : int
+            Dimension of Weyl Tensor
+        parent_metric : ~einsteinpy.symbolic.metric.WeylTensor
+            Corresponding Metric for the Weyl Tensor. Defaults to None
+
+        Raises
+        ------
+        TypeError
+            Raised when the dimension of the tensor is less than 3
+
+        """
+        if dims >= 3:
+            if dims > 3:
+                t_riemann = RiemannCurvatureTensor.from_metric(metric)
+                t_ricci = RicciTensor.from_metric(metric)
+                r_scalar = RicciScalar.from_riccitensor(t_ricci, parent_metric)
+                aux = sympy.tensorproduct(t_ricci, metric)
+                interm = sympy.tensorproduct(metric, metric)
+                # R_im g_kl
+                t1 = np.moveaxis(aux, [0, 1, 2, 3], [0, 2, 3, 1])
+                # R_il g_km
+                t2 = np.moveaxis(aux, [0, 1, 2, 3], [0, 2, 1, 3])
+                # R_kl g_im
+                t3 = np.moveaxis(aux, [0, 1, 2, 3], [2, 0, 1, 3])
+                # R_km g_il
+                t4 = np.moveaxis(aux, [0, 1, 2, 3], [2, 0, 3, 1])
+                aux1 = (t1 - t2 + t3 - t4) / (dims - 2)
```

Comment:
name `aux1` to `quantity1`, `aux` is already used, it would be confusing.
Also you would need to define `dims` somewhere above using `dims=metric.dims`